packages/edit-site/src/components/add-new-template/style.scss

```diff
@@ -178,4 +140,58 @@
 		justify-content: center;
 		height: 100%;
 	}
+	.components-modal__children-container {
```

Comment:
@ciampo just noting that it seems we have a second usage for this(if it sticks and we don't remove it).

Reply:
Is [this approach](https://github.com/WordPress/gutenberg/pull/50099#discussion_r1188628333) viable?

Reply:
No, because we care about the actual children container(the one with `components-modal__children-container`) class.

Reply:
And where is the actual height (that the child container is trying to fill) coming from...? If you have testing instructions I can look into this specific case.

If we want `.components-modal__children-container` to fill the parent height, another way would be to set `display: flex` on `.components-modal__content`. Not that it's inherently cleaner, but it could remove the need for us to keep around another overridable class 😂

(I [agree with Marco](https://github.com/WordPress/gutenberg/pull/50099#discussion_r1185900689) that in the long run we need a more modular Modal component.)

Reply:
Opened https://github.com/WordPress/gutenberg/pull/50655 with a potential approach for the Template pattern modal. Let's see if that approach can be of inspiration for this PR too

Reply:
Also, I took a look at this PR again, and I'm not understanding in which specific place is the `height: 100%` style needed — could you should me exactly what layout needs is that style fulfilling?

Reply:
Thank you Marco! The use case was the `TemplateActionsLoadingScreen` for the spinner to be placed in the middle, when we have started the creation of a template(async function with redirect afterwards).
Let's see if we'll land your [PR](https://github.com/WordPress/gutenberg/pull/50655/) and we can see this use case after that. Also it might be possible to remove that component completely.. I'll take a look.

Reply:
I updated the styles [here](https://github.com/WordPress/gutenberg/pull/50595/commits/5a939f24cc5bd200d86bfe88313ec25e8558d071) and  the internal class is not used.

Reply:
Thank you, Nik!